ballista/rust/scheduler/src/lib.rs

```diff
@@ -1001,10 +1103,56 @@ mod test {
                 namespace.to_string(),
                 BallistaCodec::default(),
             );
-        let ctx = scheduler.ctx.read().await;
-        state.init(&ctx).await?;
+        state.init().await?;
         // executor should be registered
         assert_eq!(state.get_executors_metadata().await.unwrap().len(), 1);
         Ok(())
     }
 }
+
+/// A Registry holds all the datafusion session contexts
+pub struct SessionContextRegistry {
```

Comment:
Here maybe we can add `base_context: Option<Arc<SessionContext>>`

Reply:
Hi @thinkharderdev 

Yes, this is very good point.  In the current implementation, the SessionContext is alway created using configurations from query settings/Ballista configurations, there is no way to configure different optimizers/planners.  Maybe I can add a new struct like Base SessionContext Builder and make that an extension point.

Reply:
@thinkharderdev  Done the change, please help to take a look.

Reply:
Looks great! Thanks!